source/extensions/filters/network/kafka/serialization.h

```diff
@@ -224,6 +224,72 @@ class VarUInt32Deserializer : public Deserializer<uint32_t> {
   bool ready_ = false;
 };
 
+class VarInt32Deserializer : public Deserializer<int32_t> {
+public:
+  VarInt32Deserializer() = default;
+
+  uint32_t feed(absl::string_view& data) override { return varuint32_deserializer_.feed(data); }
+
+  bool ready() const override { return varuint32_deserializer_.ready(); }
+
+  int32_t get() const override {
+    const uint32_t res = varuint32_deserializer_.get();
+    return (res >> 1) ^ -(res & 1);
+  }
+
+private:
+  VarUInt32Deserializer varuint32_deserializer_;
+};
+
+class VarInt64Deserializer : public Deserializer<int64_t> {
```

Comment:
Do we eventually need a VarUInt64Deserializer that this deserializer can be written on top off like the above?

Reply:
No, fortunately not.